binary_and_decimal.rb

```diff
@@ -3,9 +3,19 @@
 #    appears as a binary number. Calculate the decimal value for this binary
 #    number using the algorithm you devised in class. Print out the decimal value.
 def binary_to_decimal
-  puts "Not implemented"
+  binary_array = []
+  8.times do
+    binary_array << rand(0..1)
+  end
+  decimal = 0
+  8.times do |i|
+    decimal += binary_array[7-i]*(2**i)
+  end
+  return decimal
```

Comment:
Minor feedback: the problem statement asks you to print the binary and decimal values in the method. You are missing the puts statements.
The rest of the logic is good!